### PR TITLE
Editing api

### PR DESCRIPTION
**Pull request by harishdm**

### CODE REVIEW

File: lib/include/ultrahdr/editorhelper.h [hunk 1]
```diff
@@ -1,44 +0,0 @@
-/*
- * Copyright 2024 The Android Open Source Project
```
Comment (DichenZhang1) [inline lib/include/ultrahdr/editorhelper.h:2]: I would recommend that we keep this file and keep ultrahdrcommon.h simple

File: tests/editorhelper_test.cpp [hunk 2]
```diff
@@ -38,477 +94,233 @@ static bool writeFile(const char* filename, void*& result, int length) {
...
-  int out_height = IMAGE_HEIGHT * 3 / 2;
-  int outSize = out_width * out_height * 3 / 2;
-  out_img_data.reset(new uint8_t[outSize]);
-  out_img.data = out_img_data.get();
-  EXPECT_TRUE(resize(&in_img, out_width, out_height, &out_img)== JPEGR_NO_ERROR);
-  EXPECT_TRUE(out_img.width = out_width);
-  EXPECT_TRUE(out_img.height = out_height);
-  EXPECT_TRUE(out_img.colorGamut == in_img.colorGamut);
-  EXPECT_TRUE(out_img.pixelFormat == in_img.pixelFormat);
+TEST_P(EditorHelperTest, Crop) {
```
Comment (DichenZhang1) [inline tests/editorhelper_test.cpp:261]: Better to have one more test for applying multiple effects in a row, and verify the process doesn't crash, result is correct

File: lib/src/ultrahdr_api.cpp [hunk 3]
```diff
@@ -1308,3 +1391,109 @@ void uhdr_reset_decoder(uhdr_codec_private_t* dec) {
...
+    snprintf(status.detail, sizeof status.detail, "received nullptr for uhdr codec instance");
+    return status;
+  }
+
+  codec->m_effects.push_back(new ultrahdr::uhdr_mirror_effect_t(ultrahdr::UHDR_MIRROR_VERTICAL));
+
+  return status;
+}
+
+uhdr_error_info_t uhdr_add_transform_rotate90(uhdr_codec_private_t* codec) {
```
Comment (DichenZhang1) [inline lib/src/ultrahdr_api.cpp:1425]: Same

File: lib/include/ultrahdr/editorhelper.h (hunk 1 above)
Comment (vivekrj1806) [reply lib/include/ultrahdr/editorhelper.h]: Done

File: tests/editorhelper_test.cpp (hunk 2 above)
Comment (vivekrj1806) [reply tests/editorhelper_test.cpp:299]: Done

File: lib/src/ultrahdr_api.cpp [hunk 4]
```diff
@@ -1308,3 +1391,109 @@ void uhdr_reset_decoder(uhdr_codec_private_t* dec) {
     handle->m_decode_call_status = g_no_error;
   }
 }
+
+uhdr_error_info_t uhdr_add_transform_mirror_horz(uhdr_codec_private_t* codec) {
```
[the remark this answers is not in the document]
Comment (vivekrj1806) [reply lib/src/ultrahdr_api.cpp]: Done

File: lib/src/ultrahdr_api.cpp (hunk 3 above)
Comment (vivekrj1806) [reply lib/src/ultrahdr_api.cpp]: Done

**Review by DichenZhang1 — APPROVED** [review]

**Pull request merged**